components/TextFields/tests/snapshot/MDCOutlinedTextFieldSnapshotTests.m

```diff
@@ -0,0 +1,215 @@
+  UIView *dummyInputView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 0, 0)];
+  textField.inputView = dummyInputView;
+
+  // Add the textfield to the window so it's part of a valid view hierarchy and things like
+  // `-becomeFirstResponder` work.
+  UIWindow *keyWindow = [[UIApplication sharedApplication] keyWindow];
+  [keyWindow addSubview:textField];
+  return textField;
+}
+
+- (void)validateTextField:(MDCOutlinedTextField *)textField {
+  [MDCTextControlSnapshotTestHelpers validateTextControl:(UIView<MDCTextControl> *)textField
+                                            withTestCase:self];
+}
+
+#pragma mark - Tests
+
+- (void)testTextFieldWithText {
+  // Given
+  MDCOutlinedTextField *textField = self.textField;
+
+  // When
+  [MDCBaseTextFieldTestsSnapshotTestHelpers configureTextFieldWithText:textField];
+
+  // Then
+  [self validateTextField:textField];
+}
+
+- (void)testTextFieldWithLeadingView {
+  // Given
+  MDCOutlinedTextField *textField = self.textField;
+
+  // When
+  [MDCBaseTextFieldTestsSnapshotTestHelpers configureTextFieldWithLeadingView:textField];
+
+  // Then
+  [self validateTextField:textField];
+}
+
+- (void)testTextFieldWithLeadingViewWhileEditing {
+  // Given
+  MDCOutlinedTextField *textField = self.textField;
+
+  // When
+  [MDCBaseTextFieldTestsSnapshotTestHelpers
+      configureTextFieldWithLeadingViewAndTextWhileEditing:textField];
+
+  // Then
+  [self validateTextField:textField];
+}
+
+- (void)testTextFieldWithTrailingView {
+  // Given
+  MDCOutlinedTextField *textField = self.textField;
+
+  // When
+  [MDCBaseTextFieldTestsSnapshotTestHelpers configureTextFieldWithTrailingViewAndText:textField];
+
+  // Then
+  [self validateTextField:textField];
+}
+
+- (void)testTextFieldWithLeadingViewAndTrailingView {
+  // Given
+  MDCOutlinedTextField *textField = self.textField;
+
+  // When
+  [MDCBaseTextFieldTestsSnapshotTestHelpers
+      configureTextFieldWithLeadingViewAndTrailingViewAndText:textField];
+
+  // Then
+  [self validateTextField:textField];
+}
+
+- (void)testTextFieldWithVisibleClearButton {
+  // Given
+  MDCOutlinedTextField *textField = self.textField;
+
+  // When
+  [MDCBaseTextFieldTestsSnapshotTestHelpers
+      configureTextFieldWithVisibleClearButtonAndText:textField];
+
+  // Then
+  [self validateTextField:textField];
+}
+
+- (void)testFloatingLabelWithCustomColorWhileEditing {
+  // Given
+  MDCOutlinedTextField *textField = self.textField;
+
+  // When
+  [MDCBaseTextFieldTestsSnapshotTestHelpers
+      configureWithColoredFloatingLabelTextAndTextWhileEditing:textField];
+
+  // Then
+  [self validateTextField:textField];
+}
+
+- (void)testDisabledTextField {
+  // Given
+  MDCOutlinedTextField *textField = self.textField;
+
+  // When
+  [MDCBaseTextFieldTestsSnapshotTestHelpers
+      configureDisabledTextFieldWithLabelTextAndText:textField];
+
+  // Then
+  [self validateTextField:textField];
+}
+
+- (void)testEditingTextFieldWithVisiblePlaceholder {
```

Comment:
This snapshot has similar issues with the upper line.

(im ok with a todo to acknowledge and fix in a follow up PR)

Reply:
See https://github.com/material-components/material-components-ios/pull/8705#issuecomment-548894614 and https://github.com/material-components/material-components-ios/pull/8705#discussion_r341711181